tests/test_openclhpp.cpp

```diff
@@ -3626,7 +3630,63 @@ void testDevice_GetInfo_CLDeviceName()
     clGetDeviceInfo_StubWithCallback(clGetInfo_testDeviceGetInfoCLDeviceName);
     cl::string deviceName = devicePool[0].getInfo<CL_DEVICE_NAME>();
     TEST_ASSERT_EQUAL_STRING(expected.c_str(), deviceName.c_str());
-} 
+}
+
+#if defined(cl_ext_device_fission)
+static cl_int clCreateSubDevices_testDevice_createSubDevices(
+    cl_device_id in_device, const cl_device_partition_property *properties,
+    cl_uint num_devices, cl_device_id *out_devices, cl_uint *num_devices_ret,
+    int num) {
+    return 0;
+}
+
+static cl_int clCreateSubDevicesEXT_testDevice_createSubDevices(
+    cl_device_id device_in, const cl_device_partition_property_ext *properties,
+    cl_uint n, cl_device_id *out_devices, cl_uint *num, int num_calls) {
+  cl_int ret = CL_SUCCESS;
+
+  TEST_ASSERT_EQUAL(CL_DEVICE_PARTITION_TYPES_EXT, *properties);
+  if(nullptr != out_devices){
+    out_devices[0] = make_device_id(0);
+  }
+  else{
+    *num = 1;
+  }
```

Comment:
These aren't exactly real-world return values, either:

1. The number of output sub-devices should always be written to if it is non-nullptr, not only if out_devices is nullptr.
2. I don't think it's possible to partition a device into a single sub-device.

For a unit test I suppose this is OK, though it would be better to have more real-world values instead.

Reply:
Changed to always write number of sub-devices